modules/lofreq/1.1/lofreq.smk

```diff
@@ -47,15 +47,24 @@ SCRIPT_PATH = CFG['inputs']['src_dir']
 bed = str(reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.bed"))
 CFG['switches']['regions_bed']['_default'] = bed
 
-
-
 # Define rules to be run locally when using a compute cluster
 localrules:
     _lofreq_input_bam,
     _lofreq_output_vcf,
     _lofreq_all,
     _lofreq_link_to_preprocessed
 
+def _lofreq_get_capspace(sample_id, wildcards):
+
+    # If this is a genome sample, return a BED file listing all chromosomes
+    if wildcards.seq_type != "capture":
+        return CFG["switches"]["regions_bed"]
```

Comment:
Another question regarding this function (and other similar in other modules): Usually you need to define `CFG` within a function because it gives NameError of `name 'CFG' is not defined` or similar. There is even [documentation](https://lcr-modules.readthedocs.io/en/latest/faq.html#why-am-i-running-into-a-nameerror-name-cfg-is-not-defined-exception) section on it. How does it work here?

Reply:
I'm not entirely sure, but I think it's because everything in the `input` and `params` blocks are executed before `op.cleanup_module()` is called

Reply:
Yeah it's counter-intuitive but my read of the section in the docs is that functions in general will not have access to this but functions that are called by the input directive are an exception.
```
 If you use CFG in the rule directives that are evaluated when the module snakefile is parsed (e.g. input, output, log, params, etc.), it’s not an issue. However, if you use this variable in a function or run directive, i.e. code that is run after the op.cleanup_module() function is run, you will get the error above.
```

Reply:
Note to self: Code LoFreq so it will use CFG["switches"]["regions_bed"] if it exists